drivers/wdm/audio/backpln/portcls/irpstream.cpp

```diff
@@ -381,22 +411,19 @@ CIrpQueue::UpdateMapping(
     ASSERT(StreamData);
 
     // add to current offset
-    InterlockedExchangeAdd((PLONG)&m_CurrentOffset, (LONG)BytesWritten);
+    m_CurrentOffset += BytesWritten;
```

Comment:
As you've removed that call, the operation is no longer done in an atomic way to the current offset. Could you please explain the reasoning behind this?

Reply:
I am pretty sure GetMapping and UpdateMapping are always called from the same thread.

Reply:
If you look at the code for GetMapping, and UpdateMapping, they are definitely not intended to be thread safe. I made hardly any changes to those functions, they are used by wavecyclic, all the signficant changes I made were for the wavepci side. GetMappingWithTag, ReleaseMappingWithTag. CIrpQueue is kind of confusing as it implements two seperate things in the same class.